src/main/java/SpecialOffer.java

```diff
@@ -0,0 +1,23 @@
+public class SpecialOffer {
+    Product product;
+    String description;
+    int promotionTime;
+    double discount;
+
+    public SpecialOffer(Product product, String description, int promotionTime, double discount) {
+        this.product = product;
+        this.description = description;
+        this.promotionTime = promotionTime;
+        this.discount = discount;
+    }
+
+    double getDiscountPrice() {
+        return product.price - product.price * discount;
+    }
+
+    void showOfferInfo() {
+        System.out.println("Rabat na produkt 20%");
```

Comment:
Nie wiesz, czy zniżka będzie wynosiła 20%. To określamy przez wartość `discount`